pkg/content/provider/oci/oci.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oci
+
+import (
+	"archive/tar"
+	"io"
+	"path/filepath"
+	"strings"
+
+	"github.com/GoogleContainerTools/kpt/pkg/content"
+	"github.com/GoogleContainerTools/kpt/pkg/content/extensions"
+	"github.com/GoogleContainerTools/kpt/pkg/location"
+	locationmutate "github.com/GoogleContainerTools/kpt/pkg/location/mutate"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+type ociProvider struct {
+	image v1.Image
+	dir   string
+	fsys  filesys.FileSystem
```

Comment:
How is the auth setup done ? Is there a way to plug `authProvider` or the library is assuming something ?

Reply:
Yes, there's an issue filed that underlying library options will need to pass in as additional open options... 

https://github.com/google/go-containerregistry/blob/main/pkg/v1/remote/options.go

Because they're a different type, it'll be slightly nested: 

```
src, err := open.Content(ref, c.OpenOptions...) // when calculated earlier, in PreRunE
// or inline, for example
src, err := open.Content(
    ref,
    open.WithContext(ctx),
    open.WithContainerRegistryOptions(
        remote.WithAuthFromKeychain(gcrane.Keychain)),
        remote.WithTransport(etc),
        remote.WithRetryBackoff(etc),
    ),
)
```